tests/main/snapshot-users/task.yaml

```diff
@@ -0,0 +1,125 @@
+    test -e /home/test/snap/test-snapd-tools/current/canary.txt
+    test -e /home/test/snap/test-snapd-tools/common/canary.txt
+    test ! -d /home/test/snap/test-snapd-tools/current/canary
+
+    # remove the canaries for both users
+    rm -f /root/snap/test-snapd-tools/{current,common}/canary.txt
+    rm -f /home/test/snap/test-snapd-tools/{current,common}/canary.txt
+
+    # restore the snapshot for both users and check the files
+    snap restore "$SET_ID_BOTH" test-snapd-tools
+    test -e /root/snap/test-snapd-tools/current/canary.txt
+    test -e /root/snap/test-snapd-tools/common/canary.txt
+    test -e /home/test/snap/test-snapd-tools/current/canary.txt
+    test -e /home/test/snap/test-snapd-tools/common/canary.txt
+    test ! -d /home/test/snap/test-snapd-tools/current/canary
+
+    # remove the canaries for both users
+    rm -f /root/snap/test-snapd-tools/{current,common}/canary.txt
+    rm -f /home/test/snap/test-snapd-tools/{current,common}/canary.txt
+
+    # restore the snapshot for root user and check the files
+    snap restore "$SET_ID_BOTH" --users=root
+    test -e /root/snap/test-snapd-tools/current/canary.txt
+    test -e /root/snap/test-snapd-tools/common/canary.txt
+    test ! -e /home/test/snap/test-snapd-tools/current/canary.txt
+    test ! -e /home/test/snap/test-snapd-tools/common/canary.txt
+    test ! -d /home/test/snap/test-snapd-tools/current/canary
+
+    # remove the canaries for both users
+    rm -f /root/snap/test-snapd-tools/{current,common}/canary.txt
+    rm -f /home/test/snap/test-snapd-tools/{current,common}/canary.txt
+
+    # restore the snapshot for test user and check the files
+    snap restore "$SET_ID_NONE" --users=test
+    test ! -e /root/snap/test-snapd-tools/current/canary.txt
+    test ! -e /root/snap/test-snapd-tools/common/canary.txt
+    test -e /home/test/snap/test-snapd-tools/current/canary.txt
+    test -e /home/test/snap/test-snapd-tools/common/canary.txt
+    test -e /home/test/snap/test-snapd-tools/current/canary/canary.txt
+
+    # remove the canaries for both users
+    rm -f /root/snap/test-snapd-tools/{current,common}/canary.txt
+    rm -rf /home/test/snap/test-snapd-tools/{current,common,current/canary}/canary.txt
+
+    # restore the snapshot for both user and check the files
+    snap restore "$SET_ID_NONE" --users=test,root
+    test -e /root/snap/test-snapd-tools/current/canary.txt
+    test -e /root/snap/test-snapd-tools/common/canary.txt
+    test -e /home/test/snap/test-snapd-tools/current/canary.txt
+    test -e /home/test/snap/test-snapd-tools/common/canary.txt
+    test -e /home/test/snap/test-snapd-tools/current/canary/canary.txt
+
+    # check files content
+    MATCH "content updated" < /home/test/snap/test-snapd-tools/current/canary/canary.txt
+    MATCH "content updated" < /home/test/snap/test-snapd-tools/current/canary.txt
+
+    # check removal works
+    snap forget "$SET_ID_NONE"
+    snap saved --id="$SET_ID_NONE" | grep "No snapshots found"
```

Comment:
It looks like we want the same checks for the `snap forget` calls below. Maybe also a `[ $(snap saved | wc -l) = 0 ]` or something at the end?

Reply:
I implemented in that way to be consisten with the other snapshots tests, I'll see if there another better way to check that.